test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/email/RegistrationChangeEmailControllerSpec.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.email
+
+import java.util.UUID
+
+import org.scalacheck.ScalacheckShapeless._
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.i18n.MessagesApi
+import play.api.mvc.Result
+import play.api.test.CSRFTokenHelper._
+import play.api.test.FakeRequest
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.RedirectToStartBehaviour
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.RegistrationStatus.RegistrationReady
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{Email, sample}
+
+import scala.concurrent.Future
+
+class RegistrationChangeEmailControllerSpec
+  extends EmailControllerSpec[RegistrationReady,RegistrationReady]
```

Comment:
Do you think it would be worth splitting the registration and subscription controllers into separate packages?

Reply:
yeah, that would make it cleaner I think. Then we could probably get rid of `Registration` and `Subscription` from the class names - though that'd make it ambiguous if we every used those names in isolation without the correct context